python/whylogs/api/writer/mlflow.py

```diff
@@ -1,12 +1,43 @@
+import os
+from tempfile import mkdtemp
 from typing import Optional
 
+import mlflow
+
 from whylogs.api.writer import Writer
 from whylogs.core import DatasetProfileView
 
 
 class MlflowWriter(Writer):
-    def write(self, profile: DatasetProfileView, dest: Optional[str] = None) -> None:
-        pass
+    def __init__(self) -> None:
+        self._profile_dir = "whylogs"
+        self._profile_name = "whylogs_profile"
+        self._end_run = True
+
+    def write(self, profile: DatasetProfileView) -> None:
+        run = mlflow.active_run() or mlflow.start_run()
+        self._run_id = run.info.run_id
+
+        output = self._get_temp_directory(run_id=self._run_id)
+        profile.write(path=output)
+        mlflow.log_artifact(output, artifact_path=self._profile_dir)
+
+        if self._end_run:
+            mlflow.end_run()
+
+    def option(
+        self, profile_name: Optional[str] = None, profile_dir: Optional[str] = None, end_run: Optional[bool] = None
+    ) -> None:
+        if end_run:
+            self._end_run = end_run
+        if profile_dir:
+            self._profile_dir = profile_dir
+        if profile_name:
+            self._profile_name = profile_name
 
-    def option(self, param: Optional[str] = None) -> "MlflowWriter":  # type: ignore
-        pass
+    def _get_temp_directory(self, run_id):
+        tmp_dir = mkdtemp()
+        output_dir = os.path.join(tmp_dir, self._profile_dir)
+        os.makedirs(output_dir, exist_ok=True)
+        output = os.path.join(output_dir, f"{self._profile_name}_{run_id}.bin")
```

Comment:
How does the naming compare to the mlflow writer in v0? does it matter if they are different?

Reply:
Since the api changes from `mlflow.whylogs.log_pandas` to `profile.writer("mlflow").write()` I didn't think about parameters naming tbh. But it's a valid question i don't have a strict answer to. 
Could change params to `output_path, path_template, filename_template` although it's not exactly clear the difference between `output_path` and `path_template` to me. And thinking through, the other writers don't exactly follow the same naming on v0 as well. Thoughts?